packages/arb-token-bridge-ui/src/hooks/useDeposits.ts

```diff
@@ -46,16 +46,35 @@ export const fetchCompleteDepositData = async (
 
 export const useDeposits = (depositPageParams: PageParams) => {
   const { l1, l2 } = useNetworksAndSigners()
+  const { isSmartContractWallet, isLoading: isAccountTypeLoading } =
+    useAccountType()
+  const chainId = useChainId()
+
+  const isConnectedToParentChain = l1.network.id === chainId
 
   // only change l1-l2 providers (and hence, reload deposits) when the connected chain id changes
   // otherwise tx-history unnecessarily reloads on l1<->l2 network switch as well (#847)
   const l1Provider = useMemo(() => l1.provider, [l1.network.id])
   const l2Provider = useMemo(() => l2.provider, [l2.network.id])
 
   const { address: walletAddress } = useAccount()
-  const {
-    layout: { isTransactionHistoryShowingSentTx }
-  } = useAppContextState()
+
+  // sent and received txs are always displayed for EOA but not for SCW
```

Comment:
It's because SCW is tied to a specific network, so your wallet address on L1 can be someone else's wallet with the same address on L2.

So in regards to the tx history, if you are on 'To Arbitrum One' tab (deposits) and are connected to Ethereum, you should only see sent txs. If we showed both sent and received, then we can't guarantee that txs received to Arbitrum One come from your account. At this point you'd see incoming txs to someone elses account, because you only own your wallet address on Ethereum.

This complexity is non existent on EOA because you have the same wallet address across multiple networks.

I will make this comment a bit more descriptive.